src/graphnet/utils/cluster/filesys_tools.py

```diff
@@ -0,0 +1,49 @@
+"""
+Tools for file system operations
+
+Tom Stuttard
+"""
+
+import os, datetime, stat
+
+# from graphnet.utils.cluster import containers
+
+# py2 vs 3 compatibility
+try:
+    input = raw_input
+except NameError:
+    pass
+
+#
+# Tools for handling files and directories
+#
+
+# Create a directory with robust error handling, including race conditions
+def make_dir(dir_path, raise_exception=False):
+    if not os.path.exists(dir_path):
+        try:
+            os.makedirs(dir_path)
+            return True
+        except OSError:
+            assert os.path.isdir(
+                dir_path
+            ), 'Could not create directory "%s"' % (dir_path)
+    return False
```

Comment:
This seems like a lot of work, if `os.makedirs(path, exists_ok=True)` could be used, but it seems like due consideration was given to this, so I am fine to keep this if it is necessary (as it seems).